api/proto/teleport/discoveryconfig/v1/discoveryconfig.proto

```diff
@@ -50,3 +54,32 @@ message DiscoveryConfigSpec {
   // AccessGraph is the configurations for syncing Cloud accounts into Access Graph.
   types.AccessGraphSync access_graph = 6;
 }
+
+// DiscoveryConfigStatus holds dynamic information about the discovery configuration
+// running status such as errors, state and count of the resources.
+message DiscoveryConfigStatus {
+  // State reports the Discovery config state.
+  DiscoveryConfigState state = 1;
+  // error_message holds the error message when state is DISCOVERY_CONFIG_STATE_ERROR.
+  optional string error_message = 2;
+  // discovered_resources holds the count of the discovered resources in the previous iteration.
+  uint64 discovered_resources = 3;
+  // last_sync_time is the timestamp when the Discovery Config was last sync.
+  google.protobuf.Timestamp last_sync_time = 4;
+}
+
+// DiscoveryConfigState is the state of the discovery config resource.
+enum DiscoveryConfigState {
+  DISCOVERY_CONFIG_STATE_UNSPECIFIED = 0;
+  // DISCOVERY_CONFIG_STATE_RUNNING is used when the operation doesn't report
+  // incidents.
+  DISCOVERY_CONFIG_STATE_RUNNING = 1;
+  // DISCOVERY_CONFIG_STATE_ERROR is used when the operation reports
+  // incidents.
+  DISCOVERY_CONFIG_STATE_ERROR = 2;
+  // DISCOVERY_CONFIG_STATE_MISMATCHED_RESOURCE_GROUP is used when no Discovery service is configured
+  // with the discovery_group configured in this Discovery Config object.
+  DISCOVERY_CONFIG_STATE_MISMATCHED_RESOURCE_GROUP = 3;
```

Comment:
Does this need to be a separate state if we have a generic "error" state and the field for error details?